xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/displayers/linkDisplayer.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+define([
+  BASE_PATH + "displayers/defaultDisplayer.js"
+], function (
+  DefaultDisplayer
+) {
+
+
+  /**
+   * Create an HTML displayer for a property of an entry
+   * Extends the default displayer class
+   */
+  var LinkDisplayer = function (propertyId, entryData, logic) {
+    DefaultDisplayer.call(this, propertyId, entryData, logic);
+  };
+  LinkDisplayer.prototype = Object.create(DefaultDisplayer.prototype);
+  LinkDisplayer.prototype.constructor = LinkDisplayer;
+
+
+  /**
+   * Create link viewer element for the displayer
+   * Override the default inherted method
+   */
+  LinkDisplayer.prototype.createView = function (defer, params) {
+    // default href
+    var propertyHref = params.config.propertyHref && params.entry[params.config.propertyHref];
+    var href = propertyHref || params.config.href;
+    // default text
+    var text = params.value || href;
```

Comment:
I'm not sure about this. Empty value should be displayed as empty value, otherwise it may be confusing for the user.

Reply:
It was in the case there is a link but no text to display it. I thought that allowing the user to follow the link at any case was more important that disabling this ability because there are no text to bind it to. Or maybe we can display something like _(empty value)_ with the link attached to it if you prefer?

Reply:
I think it's very important that the user understands by looking at the live data entry that the property value is empty. How you achieve this visually is less important. I would certainly not display the URL though as it can take a lot of space, affecting the layout possibly, thus creating more problems than it solves.